### DISCUSSION TITLE
setup config for a specific extension

### DISCUSSION THREAD

**Discussion by jimmymjing** [airsdk/apm#83] [Ideas]
Hello, is it possible to setup required config params for a specific extension? something like `apm project config set IDFA`, and then walkthrough all required params similar to `init` command?
Similarily, is there a need to list required config params for a specific extension?

**Comment by marchbold** [comment]
Hi,

Good idea! I've been trying to think of some better approaches with adding the configuration parameters for a package. I was considering adding a similar process to the end of the install process but this sounds like a better idea, i.e. have a specific command to walkthrough the parameters for an extension.

You can currently see which extensions each parameter is specified in by using the describe command:

```
apm project config describe
```

**Comment by jimmymjing** [comment]
Cool! Do you think ppl also need to check config params for a specified extension? Something like `apm project config describe IDFA`

**Reply by marchbold** [reply]
Yeah, cool, will get something like this in too...

**Reply by marchbold** [reply]
You can check out the new `describe` params here:

https://github.com/airsdk/apm/wiki/Usage-ProjectsAndPackages#describe

Basically there are 3 variants now:

```
apm project config describe 
apm project config describe <package identifier>
apm project config describe <param name>
```

**Comment by marchbold** [comment]
The new `set` params here:

https://github.com/airsdk/apm/wiki/Usage-ProjectsAndPackages#set

The existing one was:

```
apm project config set <param name> <param value>
```

You can now do:

```
apm project config set <param name>
```

This will print the "description" and then ask for input for a value.

```
apm project config set <package identifier>
```

will print the parameters for the specified package and ask for input for each one.

Lastly,

```
apm project config set
```

(with no params) will walk through all the project parameters asking for input for each.

With any of these hitting enter with no input will leave the value unchanged.

**Reply by marchbold** [reply]
Forgot to say, this will roll out in v1.1 later this week.

**Reply by jimmymjing** [reply]
That's awesome!! Can't wait to give it a try!
